issuetracker/github/github.go

```diff
@@ -0,0 +1,35 @@
+package github
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/preslavmihaylov/todocheck/common"
+	"github.com/preslavmihaylov/todocheck/issuetracker"
+)
+
+// IssueTracker implementation for integrating with public & private github issue trackers
+type IssueTracker struct {
+	Origin string
+}
+
+// TaskModel returns the model representing a deserialized github task
+func (it *IssueTracker) TaskModel() issuetracker.Task {
+	return &Task{}
+}
+
+// TaskURLFrom taskID returns the url for the target github task ID to fetch
+func (it *IssueTracker) TaskURLFrom(taskID string) string {
+	return taskID
+}
+
+// IssueAPIOrigin returns the URL for github's issue-fetching API
+func (it *IssueTracker) IssueAPIOrigin() string {
+	tokens := common.RemoveEmptyTokens(strings.Split(it.Origin, "/"))
+	if !strings.HasPrefix(tokens[0], "http") {
+		tokens = append([]string{"https:"}, tokens...)
+	}
+
+	scheme, owner, repo := tokens[0], tokens[2], tokens[3]
```

Comment:
I always try to avoid using direct access with indexes since it's so easy to face an `index out of range` error and hard to maintain in the future.
Instead, we can use `regexp` for example.

I suggest moving the part of extracting data from `it.Origin` to a private method as well.

Reply:
I agree that using `regexp` would be much better here. To avoid making too many changes in a single PR, would you agree to file an issue for doing that instead?

As for extracting it to a private function, I think that would be unnecessary given that the function is ~8 LOC long. If we do that, then this public function will simply become a wrapper. What do you think?

Reply:
Of course, let's go with an issue for later. I like to keep the PRs small an simple as well.

About the private method, although I don't consider a public method which is only a wrapper for some other private methods a bad practice, I think we can stick we this for now since it's tiny as you mentioned.